packages/sqlite/src/index.ts

```diff
@@ -118,115 +92,91 @@ class DatabaseConnectionImplementation implements DatabaseConnection {
 
   async tx<T>(fn: (db: DatabaseTransaction) => Promise<T>): Promise<T> {
     return this._mutex.writeLock(async () => {
-      await new Promise<void>((resolve, reject) => {
-        this._database.run('BEGIN', (err) => {
-          if (err) reject(err);
-          else resolve();
-        });
-      });
+      // TODO extract those as class variables
+      // however I do not know how to get
+      this._begin.run();
       try {
-        const result = fn(
+        const result = await fn(
           new DatabaseTransactionImplementation(this._database),
         );
-        await new Promise<void>((resolve, reject) => {
-          this._database.run('COMMIT', (err) => {
-            if (err) reject(err);
-            else resolve();
-          });
-        });
+        this._commit.run();
         return result;
       } catch (ex) {
-        await new Promise<void>((resolve, reject) => {
-          this._database.run('ROLLBACK', (err) => {
-            if (err) reject(err);
-            else resolve();
-          });
-        });
+        this._rollback.run();
         throw ex;
       }
     });
   }
 
   async dispose() {
-    await new Promise<void>((resolve, reject) => {
-      this._database.close((err) => {
-        if (err) reject(err);
-        else resolve();
-      });
-    });
+    this._database.close();
   }
 }
 
 export default function connect(
   filename: string = IN_MEMORY,
-  options: DatabaseConnectionOptions = {},
+  options: DatabaseOptions = {},
 ): DatabaseConnection {
   return new DatabaseConnectionImplementation(filename, options);
 }
 module.exports = Object.assign(connect, {
   default: connect,
-  DatabaseConnectionMode,
   IN_MEMORY,
   sql,
   isSqlQuery,
 });
 
 async function runQuery(
   query: SQLQuery,
-  database: sqlite.Database,
+  database: Database,
   lock: <T>(fn: () => Promise<T>) => Promise<T>,
 ) {
   const {text, values} = query.format(sqliteFormat);
   return lock(async () => {
-    return await new Promise<any[]>((resolve, reject) => {
-      database.all(text, values, (err, rows) => {
-        if (err) reject(err);
-        else resolve(rows);
-      });
-    });
+    const stm = database.prepare(text);
```

Comment:
Does `better-sqlite3` automatically cache/deduplicate calls to `database.prepare` or do we need to do that in our driver? I would imagine this could be pretty significant for performance.

Reply:
I don't think `better-sqlite3` does cache them. Should we?